Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2463,3 +2463,37 @@ extension BrowserViewController: DevicePickerViewControllerDelegate, Instruction
     }
 }
 
+
+// MARK: - reopen last closed tab
+
+extension BrowserViewController {
+    
+    override func motionEnded(_ motion: UIEventSubtype, with event: UIEvent?) {
+        homePanelDidRequestToRestoreClosedTab(motion)
+    }
+    
+    func homePanelDidRequestToRestoreClosedTab(_ motion: UIEventSubtype) {
+        guard motion == .motionShake, !topTabsVisible, !urlBar.inOverlayMode,
+            let selectedTab = tabManager.selectedTab,
+            let selectedURL = selectedTab.url,
+            let lastClosedURL = profile.recentlyClosedTabs.tabs.first?.url,
+            InternalURL.isValid(url: selectedURL),
+            selectedURL.path.starts(with: "/\(AboutHomeHandler.path)") else { return }
```

Comment:
If I'm reading this correctly, this means that the shake motion only works when you are on a new tab. I think this is confusing and won't be as clear to the user when the shake UI is available and when it isnt. Is there a specific reason why you added this?

Reply:
Thus spoke title #4725: "...should be an option on __new__ tab page".

Reply:
@akarataev Sorry the description of the issue was a bit confusing on my part. It was auto imported from bugzilla and not actually written by me. I also spoke to @bryanbell who confirmed that this should happen regardless of if you are on the new tab. It'll be too confusing otherwise. People might not quickly grok that you need to be on a new tab page to trigger this UI.

Reply:
@farhanpatel well, I already fixed this when updating PR.

Reply:
sorry! didnt see that. Merging now